apps/rig_tests/test/proxy/request_logger/kafka_test.exs

```diff
@@ -0,0 +1,185 @@
+defmodule RigTests.Proxy.RequestLogger.KafkaTest do
+  @moduledoc """
+  When API Gateway request arrives, event is published to Kafka topic.
+  """
+  use Rig.Config, [
+    :brokers,
+    :consumer_topics,
+    :ssl_enabled?,
+    :ssl_ca_certfile,
+    :ssl_certfile,
+    :ssl_keyfile,
+    :ssl_keyfile_pass,
+    :sasl
+  ]
+
+  use ExUnit.Case, async: false
+
+  import FakeServer
+
+  alias FakeServer.Response
+  alias Rig.KafkaConfig, as: RigKafkaConfig
+  alias RigKafka
+
+  @api_port Confex.fetch_env!(:rig_api, RigApi.Endpoint)[:http][:port]
+  @proxy_port Confex.fetch_env!(:rig_inbound_gateway, RigInboundGatewayWeb.Endpoint)[:http][:port]
+  @env [port: 55_001]
+
+  defp kafka_config, do: RigKafkaConfig.parse(config())
+
+  setup do
+    System.put_env("REQUEST_LOG", "kafka")
+    kafka_config = kafka_config()
+
+    test_pid = self()
+
+    callback = fn
+      msg ->
+        send(test_pid, msg)
+        :ok
+    end
+
+    {:ok, kafka_client} = RigKafka.start(kafka_config, callback)
+
+    on_exit(fn ->
+      RigKafka.Client.stop_supervised(kafka_client)
+    end)
+
+    :ok
+  end
+
+  @tag :kafka
+  test_with_server "Given request logger is set to Kafka, the http request should publish message to Kafka topic",
+                   @env do
+    test_name = "proxy-logger-kafka"
+
+    api_id = "mock-#{test_name}-api"
+    endpoint_id = "mock-#{test_name}-endpoint"
+    endpoint_path = "/#{endpoint_id}"
+
+    # We register the endpoint with the proxy:
+    rig_api_url = "http://localhost:#{@api_port}/v1/apis"
+    rig_proxy_url = "http://localhost:#{@proxy_port}"
+
+    route(endpoint_path, Response.ok!(~s<{"status":"ok"}>))
+
+    body =
+      Jason.encode!(%{
+        id: api_id,
+        name: "Mock API",
+        version_data: %{
+          default: %{
+            endpoints: [
+              %{
+                id: endpoint_id,
+                method: "POST",
+                path: endpoint_path
+              }
+            ]
+          }
+        },
+        proxy: %{
+          target_url: "localhost",
+          port: 55_001
+        }
+      })
+
+    headers = [{"content-type", "application/json"}]
+    HTTPoison.post!(rig_api_url, body, headers)
+
+    # The client calls the proxy endpoint:
+    request_url = rig_proxy_url <> endpoint_path
+
+    :timer.sleep(5_000)
+
+    %HTTPoison.Response{status_code: res_status, body: res_body} =
+      HTTPoison.post!(request_url, "", headers)
+
+    assert res_status == 200
+    assert res_body == "{\"status\":\"ok\"}"
+
+    assert_receive received_msg, 10_000
+    received_msg_map = Jason.decode!(received_msg)
+
+    System.delete_env("REQUEST_LOG")
+
+    # Endpoint
+    assert get_in(received_msg_map, ["data", "endpoint", "id"]) == endpoint_id
```

Comment:
I wonder if it'd make sense to have a type field of some sort that tells us whether this is actually a log entry or just some json that happens to have the same fields. What do you think?

Reply:
You mean cloud events' type?

Reply:
of course, didn't really consider that this is _already_ a cloudevent ^^ yes, the cloudevent's type seems appropriate